tests/testthat/test-shinytest2-tm_a_pca.R

```diff
@@ -0,0 +1,171 @@
+testthat::test_that("e2e - tm_a_pca: data extract changes output", {
+  skip_if_too_deep(5)
+
+  require(shinytest2)
+
+  data <- within(teal_data(), {
+    require(nestcolor)
+    require(ggplot2)
+
+    USArrests <- USArrests # nolint: object_name.
+  })
+  datanames(data) <- "USArrests"
+
+  app <- TealAppDriver$new(
+    data = data,
+    modules = tm_a_pca(
+      dat = data_extract_spec(
+        dataname = "USArrests",
+        select = select_spec(
+          choices = variable_choices(
+            data = data[["USArrests"]],
+            c("Murder", "Assault", "UrbanPop", "Rape")
+          ),
+          selected = c("Murder", "Assault"),
+          multiple = TRUE
+        )
+      )
+    )
+  )
+
+  # Data selection (adds rows to tables)
+  app$set_module_input("dat-dataset_USArrests_singleextract-select", c("Murder", "Assault"))
```

Comment:
I get a warning after a timeout when we try to set this input as `Murder + Assault` because it is already selected when the module is initialised.
Perhaps, we need to just check if this input is set to this value instead of setting it?

Reply:
This can be removed and we just check if the initial screen has no errors

Reply:
Thanks for adding `wait = FALSE`. Resolving it now as it does not block anymore.